src/topp/GNPSExport.cpp

```diff
@@ -0,0 +1,358 @@
+						feature_stream << "SCANS=" << to_string(i) << endl;
+
+						string filename = mzml_file_paths[peptide.second.first];
+						Size parse_index = filename.rfind("/") + 1;
+						filename = filename.substr(parse_index);
+						feature_stream << "FILENAME=" << filename << endl;
+						feature_stream << "CONSENSUSID=e_" << feature.getUniqueId() << endl;
+
+						feature_stream << "MSLEVEL=2" << endl;
+						feature_stream << "CHARGE=" << to_string(charge == 0 ? 1 : charge) << "+" << endl;
+						feature_stream << "PEPMASS=" << peptide.first.second.getMZ() << endl;
+						feature_stream << "FILE_INDEX=" << peptide.second.second << endl;
+						feature_stream << "RTINSECOND=" << peptide.first.second.getRT() << endl;
+
+						auto ms2_scan = ms_maps[peptide.second.first][peptide.second.second];
+						// sort spectra
+						sort(ms2_scan.begin(), ms2_scan.end(), [](const Peak1D& a, const Peak1D& b) {
+							return a.getMZ() > b.getMZ();
+						});
+						// ms2_scan.sortByIntensity(true);
+
+						for(Size l = 0; l < ms2_scan.size(); l++) {
+							feature_stream << ms2_scan[l].getMZ() << "\t" << to_string(ms2_scan[l].getIntensity()) << endl;
+						}
+
+						feature_stream << "END IONS" << endl << endl;
+					}
+					feature_count++;
+				} else { // merged spectra
+					// map mz to intensity
+					map<double,int> ms2_block;
+
+					// MSExperiment exp;
+
+					const BinnedSpectrum binned_highest_int(ms_maps[peptides[0].second.first][peptides[0].second.second], BinnedSpectrum::DEFAULT_BIN_WIDTH_HIRES, false, 1, BinnedSpectrum::DEFAULT_BIN_OFFSET_HIRES);
+
+					for(auto peptide : peptides) {
+						int map_index = peptide.second.first;
+						int spectra_index = peptide.second.second;
+						// int highest_binned_intensity = peptide.first.first;
+						// auto highest_peptide_identification = peptide.first.second;
+
+						auto spectrum = ms_maps[map_index][spectra_index];
+
+						const BinnedSpectrum binned_spectrum(spectrum, BinnedSpectrum::DEFAULT_BIN_WIDTH_HIRES, false, 1, BinnedSpectrum::DEFAULT_BIN_OFFSET_HIRES);
+
+						BinnedSpectralContrastAngle bsca;
+						double cosine_sim = bsca(binned_highest_int, binned_spectrum);
+						// LOG_DEBUG << cosine_sim << " >= " << cos_sim << endl;
+
+						// compare calculated cosine sim to binned highest int
+						if(cosine_sim >= cos_sim) {
+							for(Size spectrum_index = 0; spectrum_index < spectrum.size(); ++spectrum_index) {
+								// exp.addSpectrum(spectrum);
+								// exp.addSpectrum(spectrum);
+								auto curr_spectrum = spectrum[spectrum_index];
+								if (ms2_block[curr_spectrum.getMZ()] < curr_spectrum.getIntensity()) {
+									ms2_block[curr_spectrum.getMZ()] = curr_spectrum.getIntensity();
+								}
+							}
+						}
+					}
+
+					// SpectraMerger merger;
+					// Param p;
+					// p.setValue("precursor_method:mz_tolerance", prec_mz_tol);
+					// p.setValue("precursor_method:rt_tolerance", prec_rt_tol*2);
```

Comment:
How do you merge the spectra in your merged spectra section (e.g. small comment)?

Could you please remove unused commented code (e.g.  SpectraMerger).